### ISSUE TITLE
[BUG] Code Coverage Report empty

### ISSUE THREAD

**Issue by webreidi** [microsoft/vscode-dotnettools#1458]
### Describe the Issue

 Originally submitted in https://github.com/microsoft/vscode-dotnettools/issues/1035 via @ChristoWolf
Yes, the option shows and executing it leads to the following:

The C# Dev Kit - Code Coverage output reads:
Created Coverage Controller
CodeCoverage: Report generated: <redacted>
The C# Dev Kit - Test Explorer output reads:
Using custom runsettings file at: <redacted>
========== Starting test run ==========
MSTest Executor: Test Parallelization enabled for 
<redacted> (Workers: 24, Scope: MethodLevel).
========== Test run finished: 1 Tests (1 Passed, 0 Failed, 0 Skipped) run in 150.4 ms ==========
So nothing unusual so far I'd say.
However, only 0% are shown for all covered files:
image
Could it have something to do with the coverage collector that I am using (<PackageVersion Include="coverlet.collector" Version="6.0.0" />)?

**Comment by webreidi** [comment]
@fhnaseer opened a new issue to track this under instead of under the old enhancement request.

**Comment by ChristoWolf** [comment]
Because it got lost: I am using the coverlet.collector v6.0.0.

**Comment by suraciii** [comment]
I met the save problem

Run `dotnet test --collect:"Code Coverage"`, and open the result file in vscode by `.NET: Open Coverage Report...` command, vscode can display coverage correctly

Run test with coverage in vscode, then open the result file in `*\AppData\Local\Temp\1547632658` (which was logged in 'C# Dev Kit - Code Coverage' window), vscode display broken coverage results

a snippet of logs 
[F5044E21-Microsoft.CodeCoverage.VSCode.CollectionService-22960-iygokcqb-1.log](https://github.com/user-attachments/files/17001706/F5044E21-Microsoft.CodeCoverage.VSCode.CollectionService-22960-iygokcqb-1.log)

**Comment by ChristoWolf** [comment]
I tried to reproduce this at home using a minimal solution with library and test project, but was not able to.

Not sure how to best proceed with this.
Is there an option to increase any output?

**Comment by fhnaseer** [comment]
> I tried to reproduce this at home using a minimal solution with library and test project, but was not able to.
> 
> Not sure how to best proceed with this. Is there an option to increase any output?



> I met the save problem
> 
> Run `dotnet test --collect:"Code Coverage"`, and open the result file in vscode by `.NET: Open Coverage Report...` command, vscode can display coverage correctly
> 
> Run test with coverage in vscode, then open the result file in `*\AppData\Local\Temp\1547632658` (which was logged in 'C# Dev Kit - Code Coverage' window), vscode display broken coverage results
> 
> a snippet of logs [F5044E21-Microsoft.CodeCoverage.VSCode.CollectionService-22960-iygokcqb-1.log](https://github.com/user-attachments/files/17001706/F5044E21-Microsoft.CodeCoverage.VSCode.CollectionService-22960-iygokcqb-1.log)

@suraciii @ChristoWolf If you are able to repro this on a minimal solution (library and test project), then can you please share the minimal solution and coverage file? Coverage file location can be found in the log file `AppData\Local\Temp\number`

**Comment by ChristoWolf** [comment]
> I met the save problem
> 
> Run `dotnet test --collect:"Code Coverage"`, and open the result file in vscode by `.NET: Open Coverage Report...` command, vscode can display coverage correctly
> 
> Run test with coverage in vscode, then open the result file in `*\AppData\Local\Temp\1547632658` (which was logged in 'C# Dev Kit - Code Coverage' window), vscode display broken coverage results
> 
> a snippet of logs [F5044E21-Microsoft.CodeCoverage.VSCode.CollectionService-22960-iygokcqb-1.log](https://github.com/user-attachments/files/17001706/F5044E21-Microsoft.CodeCoverage.VSCode.CollectionService-22960-iygokcqb-1.log)

It's exactly the same for me.

Unfortunately, I was not able to reproduce this with a minimal solution yet.

**Comment by ChristoWolf** [comment]
I also tried to open my problematic solution in a dedicated VS Code instance to check if my huge workspace has anything to do with it, but to no avail, coverage is also broken there.

**Comment by fhnaseer** [comment]
@ChristoWolf Can you please share coverage file generated against your solution? Name should be in the log file (as I mentioned in my previous comment). You can share here or send it to fhnaseer@live.com.

**Comment by ChristoWolf** [comment]
@fhnaseer I cannot do that unfortunately due to company policy.

**Comment by ChristoWolf** [comment]
@fhnaseer I completely forgot that I could do that, thanks!

When I open the report in Visual Studio Enterprise, I see the correct projects, but all of them have 0% coverage.

When exported to xml, my modules show under `modules` and nothing is skipped, but they already show as not covered (coverages are all 0).

**Comment by fhnaseer** [comment]
@ChristoWolf If you collect coverage from Visual Studio Enterprise, do you see same coverage results (modules included and zero coverage)?

**Comment by ChristoWolf** [comment]
@fhnaseer Yes, when computed from Visual Studio, the code coverage is correctly yielded and visualized.

I am using Microsoft Visual Studio Enterprise 2022 (64-bit) - Current, Version 17.11.3.

**Comment by fhnaseer** [comment]
Do you have 32bit version of VSCode? Are you running tests under 32bit?

**Comment by ChristoWolf** [comment]
My VS Code is 64 bit, v1.93.1.

**Comment by fhnaseer** [comment]
I fixed some issues which will be available in the next release. Since I am not able to repro your issue, so I cannot guarantee that these will fix your issue.

> Unfortunately, I was not able to reproduce this with a minimal solution yet.


> I also tried to open my problematic solution in a dedicated VS Code instance to check if my huge workspace has anything to do with it, but to no avail, coverage is also broken there.

Can you please share target framework version? Are you using Fakes?
Some suggestions that you can try to help in figuring out main problem,
1. In your unit test project
1.1. Comment all tests and remove all project references
1.2. Add a new library project and add unit tests for it and check for coverage
1.3. If it works, then add some code/project back and try again
1.4. if it fails, then try changing target framework

**Comment by fhnaseer** [comment]
Closing this as duplicate of https://github.com/microsoft/vscode-dotnettools/issues/1779.

The issue occurs if VSCode is running in admin mode. This has been fixed and should be available in the next milestone.
